pkg/server/plugin/datastore/sql/postgres.go

```diff
@@ -2,17 +2,24 @@ package sql
 
 import (
 	"github.com/jinzhu/gorm"
+
 	// gorm postgres dialect init registration
 	_ "github.com/jinzhu/gorm/dialects/postgres"
 	_ "github.com/spiffe/spire/pkg/server/plugin/datastore"
 )
 
-type postgres struct{}
+type postgresDB struct{}
 
-func (p postgres) connect(cfg *configuration, isReadOnly bool) (*gorm.DB, error) {
+func (p postgresDB) connect(cfg *configuration, isReadOnly bool) (*gorm.DB, string, bool, error) {
 	db, err := gorm.Open("postgres", getConnectionString(cfg, isReadOnly))
 	if err != nil {
-		return nil, sqlError.Wrap(err)
+		return nil, "", false, sqlError.Wrap(err)
 	}
-	return db, nil
+
+	version, err := queryVersion(db, "SHOW server_version")
+	if err != nil {
+		return nil, "", false, err
+	}
+
+	return db, version, true, nil
```

Comment:
Does this mean postgres has always supported CTE? That all the versions _we_ support support it? A comment or something could be useful context. Same for sqlite

Reply:
CTE is supported for all of the versions of postgres that we support (9 through 11). CTE support is postgres goes back even farther though (at least 8).

CTE is supported by the version of sqlite3 that we build into SPIRE.

I'll update the comments.